http3/frame_type.go

```diff
@@ -0,0 +1,52 @@
+package http3
+
+import "fmt"
+
+const (
+	FrameTypeData          FrameType = 0x00
+	FrameTypeHeaders       FrameType = 0x01
+	FrameTypeCancelPush    FrameType = 0x03
+	FrameTypeSettings      FrameType = 0x04
+	FrameTypePushPromise   FrameType = 0x05
+	FrameTypeGoAway        FrameType = 0x07
+	FrameTypeMaxPushID     FrameType = 0x0d
+	FrameTypeDuplicatePush FrameType = 0x0e
+
+	// https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-01.html#section-7.3
+	FrameTypeWebTransportStream FrameType = 0x41
+
+	// https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-03.html#section-10.1
+	FrameTypeCapsule FrameType = 0xffcab5
+)
+
+// A FrameType represents an HTTP/3 frame type.
+// https://www.ietf.org/archive/id/draft-ietf-quic-http-34.html#name-frame-definitions
+type FrameType uint64
```

Comment:
Why is this exported?

Reply:
`FrameType` constants are (IETF) spec-defined values which can be used by callers implementing parts of HTTP/3 directly.